tests/simulator_test.cpp

```diff
@@ -132,25 +122,28 @@ TEST_F(EnergyCollectorTest, NotEvenNumOfEnergyCollectorTest) {
 
   Ray alongX(kVecZero, kVecX);
   ASSERT_TRUE(performHitCollector(energyCollectors, alongX, &hitData));
-  ASSERT_EQ(collectorPositionRadius - refCollectorRadius, hitData.time);
+  ASSERT_FLOAT_EQ(collectorPositionRadius - refCollectorRadius, hitData.time);
 
   Ray alongY(kVecZero, kVecY);
   ASSERT_TRUE(performHitCollector(energyCollectors, alongY, &hitData));
-  ASSERT_EQ(collectorPositionRadius - refCollectorRadius, hitData.time);
+  ASSERT_FLOAT_EQ(collectorPositionRadius - refCollectorRadius, hitData.time);
 
-  Ray at2Angle(kVecZero, Vec3(std::cos(2 * collectorAngle), 0,
-                              std::sin(2 * collectorAngle)));
+  const float collectorAngle = 2 * kPi / (numCollectors - 1);
+  Ray at2Angle(kVecZero, Vec3(std::cos(0.5 * collectorAngle), 0,
+                              std::sin(0.5 * collectorAngle)));
   ASSERT_TRUE(performHitCollector(energyCollectors, at2Angle, &hitData));
   // TODO: This case doesn't work, find out why
-  // ASSERT_EQ(collectorPositionRadius - refCollectorRadius, hitData.time);
+  // printCollectors(energyCollectors);
+  // ASSERT_FLOAT_EQ(collectorPositionRadius - refCollectorRadius, hitData.time);
 
   Ray at2AngleOther(kVecZero, Vec3(-std::cos(2 * collectorAngle), 0,
                                    std::sin(2 * collectorAngle)));
   std::cout << at2AngleOther << std::endl;
 
   ASSERT_TRUE(performHitCollector(energyCollectors, at2AngleOther, &hitData));
   // TODO: same shit
-  // ASSERT_EQ(collectorPositionRadius - refCollectorRadius, hitData.time);
+  // ASSERT_FLOAT_EQ(collectorPositionRadius - refCollectorRadius,
```

Comment:
What is the time you are getting?

Reply:
3.8784485 ... which is so weird based on thefact that simulation radius is equal to 4.